packages/venia-concept/src/components/ProductFullDetail/__tests__/ProductFullDetail.spec.js

```diff
@@ -95,11 +96,9 @@ test('Configurable Product has correct media gallery image count', async () => {
         </WindowSizeContextProvider>
     );
 
-    const { instance } = root.children[0].children[0];
+    const productFullDetailComponent = root.children[0].children[0];
+    const carouselComponent =
+        productFullDetailComponent.children[0].children[1].children[0];
 
-    expect(instance.mediaGalleryEntries).toHaveLength(2);
-    instance.setState({
-        optionSelections: new Map([['1', 1]])
-    });
-    expect(instance.mediaGalleryEntries).toHaveLength(3);
+    expect(carouselComponent.props.images).toHaveLength(2);
```

Comment:
We did lose an assertion here (the result of updating options) but since these components are functions as opposed to classes now, we don't have access to an `instance` on which to set `state`.

Reply:
There is probably a way to do this but it'll involve mocking `useState` or somehow gaining access to the `setQuantity` callback and then using the that function within an `act`. Look at some of the useQueryResult.spec.js tests for inspiration :)

Reply:
Resolved in https://github.com/magento-research/pwa-studio/pull/1240/commits/cf77f6f1f14c5164b902ad0ed10cdda1cba0face.